### ISSUE TITLE
Open to adding a time machine size option?

### ISSUE THREAD

**Issue by chmac** [dperson/samba#295]
Firstly, thanks a lot for this samba container. It was the final piece in my long journey to get time machine to backup to an ubuntu server! :-)

I'm looking at `samba.sh` and wondering if it would be practical to add an option for the samba share setting `fruit:time machine max size`. What do you think?

I had a look at the arguments and it's not clear to me how this option would fit, but maybe it could be added to the sequence somehow. I'm happy to look at creating a PR if you're open to the change, although my bash-fu is pretty weak... :-)

**Comment by dperson** [comment]
I think what you're looking for is the `-g` option, it's in `README.md`.

**Comment by chmac** [comment]
@dperson I think the `-g` option is for global flags. The time machine max size option can be both, and in this case I want to set it per share. I’m aiming to host multiple time machine backups with a single samba server, and ideally each with their own size limit.

**Comment by dperson** [comment]
The share creation is already overly complex for my taste, unless you can come up with a graceful way to add it to specific shares, I'm going to pass. Sorry.

**Comment by chmac** [comment]
Makes sense. I'll look into it a bit and maybe whip up a fork, or some kind of [patch-package](https://www.npmjs.com/package/patch-package) type approach if I can find a way.

Thanks for making samba accessible to us regular humans! :-)

**Comment by vertoforce** [comment]
@chmac Are you using this image to host a mac time machine? If so, how did you set that up?

@dperson I saw some previous conversations (https://github.com/dperson/samba/issues/176 https://github.com/dperson/samba/issues/133) about time machine support, was there ever any progress on that front?  I'm interested in using the share for time machine.

**Comment by vertoforce** [comment]
Hey if anybody is looking to do the same, I got it working with this configuration:

```yml
command: '-g "fruit:aapl = yes" -g "vfs objects = catia fruit streams_xattr" -g "fruit:time machine = yes" -g "durable handles = yes" -g "kernel oplocks = no" -g "kernel share modes = no" -g "posix locking = no" -g "fruit:advertise_fullsync = true" -g "smb2 leases = yes" -s "your share;/media/;yes;no;no;bob;none" -u "bob;bobpasswd" -p'
```

Some stuff is probably not necessary to be honest

**Comment by chmac** [comment]
@vertoforce Glad you got it working. I **was** using it to host a time machine backup. I found that the setup was so unreliable, that I dropped it in the end. Certainly can't fault @dperson's docker image for that, it seems that MacOS and Samba just don't place super nicely for Time Machine backups. Eventually after a year or so of backups, with a few recurring errors, finally Time Machine told me that I needed to recreate the backup from scratch as it had become corrupted. Read a bit about potential fixes, and finally decided to buy more external hard drives and move on with my life.
